htdocs/include/ajax/mentions.php

```diff
@@ -0,0 +1,92 @@
+<?php
+declare(strict_types=1);
+
+// ImpressCMS AJAX endpoint for CKEditor 4 Mentions plugin
+// Returns a JSON array of users: [{ id, name, username, avatar }]
+
+// Bootstrap ImpressCMS
+require_once dirname(__DIR__, 2) . '/mainfile.php';
+
+// Helpers
+function jsonOut($data, int $status = 200): void {
+    http_response_code($status);
+    header('Content-Type: application/json; charset=utf-8');
+    echo json_encode($data, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+    exit;
+}
+
+// CSRF validation (expects XOOPS_TOKEN_REQUEST in GET/POST)
+if (!icms::$security->validateToken(false, false)) {
+    jsonOut(['error' => 'Invalid CSRF token'], 403);
+}
+
+// Only allow GET
+if (strtoupper($_SERVER['REQUEST_METHOD'] ?? 'GET') !== 'GET') {
+    jsonOut(['error' => 'Method not allowed'], 405);
+}
+
+// Read and sanitize query params
+$q = isset($_GET['q']) ? trim((string) $_GET['q']) : '';
+$limit = isset($_GET['limit']) ? max(1, min(50, (int) $_GET['limit'])) : 50; // cap at 50
+
+// Early return on empty query (mentions plugin may call with empty query after marker)
+if ($q === '') {
+    jsonOut([]);
+}
+
+// Build search (active users only)
+try {
+    /** @var PDO $pdo */
+    $pdo = icms::$db; // PDO connection
+
+    $sql = 'SELECT `uid`, `uname`, `name`, `user_avatar`, `email` FROM `' . str_replace('`', '', icms::$xoopsDB->prefix('users')) . '`
+            WHERE (`uname` LIKE :q OR `name` LIKE :q) AND `level` > 0
+            ORDER BY `uname` ASC
+            LIMIT :lim';
+
+    $stmt = $pdo->prepare($sql);
+    $like = '%' . $q . '%';
+    $stmt->bindValue(':q', $like, PDO::PARAM_STR);
+    $stmt->bindValue(':lim', $limit, PDO::PARAM_INT);
+    $stmt->execute();
```

Comment:
We have IPF... so why not use it? I think this code should be in User handler asone of method and here called.

Reply:
Good point. The prepared statement stuff is something that I would like to take over in the IPF base classes.

Reply:
I replaced this with a IPF request as well.